src/library/utils/children.js

```diff
@@ -2,6 +2,23 @@
 import { Children } from 'react';
 import { toArray } from './collections';
 
+const hasChildren = (child: React$Element<*>): boolean =>
+  Boolean(child && child.props && child.props.children);
+
+const hasComplexChildren = (child: React$Element<*>): boolean =>
+  hasChildren(child) && typeof child.props.children === 'object';
```

Comment:
Is `typeof child.props.children === 'object'` a safe (and clear) enough check here, given that so many JS entities distill down to objects?

Reply:
Good question.  I believe so.  `hasChildren` handles the `null` check and this essentially checks it is not a `string`, `number`, `boolean`.